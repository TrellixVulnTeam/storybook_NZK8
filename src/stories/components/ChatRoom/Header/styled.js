// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';
import backIcon from './img/backIcon.svg';

export const Wrapper = styled.div`
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  padding: 10px 20px;
  border-bottom: 1px solid ${colors.paleGrey};
`;

export const Details = styled.div`
  padding-left: 14px;

  label {
    display: block;
  }

  label:first-child {
    font-size: 16px;
    font-weight: 600;
    color: ${colors.slateGrey};
  }

  label:last-child {
    font-size: 12px;
    color: ${colors.steel};
  }
`;

export const Options = styled.div`
  flex: 1;
  text-align: right;

  > div {
    cursor: pointer;
  }
`;

export const BackIcon = styled(backIcon)`
  width: 20px;
  height: 20px;
  margin-right: 10px;

  &:hover {
    cursor: pointer;
  }

  @media (min-width: 700px) {
    display: none;
  }
`;
