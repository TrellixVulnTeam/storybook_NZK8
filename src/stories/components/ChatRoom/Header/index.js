// @flow

import React from 'react';
import get from 'lodash.get';
import Tooltip from 'components/Tooltip';
import type { RouterHistory } from 'react-router';
import { Wrapper, Details, Options, BackIcon } from './styled';
import { Participants, Avatar } from '../../ChannelListItem/styled';
import TeamsIcon from './img/teamIcon.svg';

type Props = {
  // eslint-disable-next-line react/no-unused-prop-types
  channel?: ?Object,
  session: Object,
  history: RouterHistory,
  onOpenTeamPanel: Function,
};

export default (props: Props) => {
  const users = get(props, 'channel.users', []);
  const description = get(props, 'channel.description', null);
  const type = get(props, 'channel.type', null);
  const otherUsers = users.filter(a => a.id !== props.session.id);

  let name = get(props, 'channel.name', null);
  let participantContent = <Participants>{users.length ? users.length : null}</Participants>;
  if (type === 'DIRECT_MESSAGE' && otherUsers.length) {
    if (otherUsers[0].firstName && otherUsers[0].lastName) {
      name = `${otherUsers[0].firstName} ${otherUsers[0].lastName}`;
    } else {
      name = otherUsers[0].email;
    }
    participantContent = <Avatar src={otherUsers[0].avatar || '/static/web/assets/defaultAvatar.png'} />;
  }

  return (
    <Wrapper>
      <BackIcon onClick={() => props.history.push('/web/messages')} />
      {participantContent}
      <Details>
        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
        <label>{name}</label>
        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
        <label>{description}</label>
      </Details>
      <Options>
        <Tooltip value="Team">
          {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */}
          <span onClick={props.onOpenTeamPanel}>
            <TeamsIcon />
          </span>
        </Tooltip>
      </Options>
    </Wrapper>
  );
};
