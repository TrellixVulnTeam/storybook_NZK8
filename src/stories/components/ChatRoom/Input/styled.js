// @flow

import styled, { keyframes } from 'styled-components';
import { colors } from 'utils/StyleGuide';
import paperclip from './img/paperclip.svg';
import emojiIconActive from './img/emojiIconActive.svg';
import emojiIconInactive from './img/emojiIconInactive.svg';

export const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  border-top: 1px solid ${colors.paleGrey};
`;

export const TextArea = styled.textarea`
  flex: 1;
  font-size: 14px;
  color: ${colors.charcoalGrey};
  padding: 20px 45px;
  border: none;
  outline: none;
  resize: none;
  overflow: hidden;

  ::placeholder {
    color: ${colors.steel};
  }
`;

export const FileButton = styled.div`
  position: absolute;
  left: 10px;
  overflow: hidden;
  cursor: pointer;

  input[type='file'] {
    position: absolute;
    top: 0;
    right: 0;
    min-width: 100%;
    min-height: 100%;
    font-size: 100px;
    text-align: right;
    filter: alpha(opacity=0);
    opacity: 0;
    outline: none;
    background: white;
    cursor: inherit;
    display: block;
  }
`;

export const PaperclipIcon = styled(paperclip)`
  width: 25px;
  height: 25px;
`;

export const ThumbnailContainer = styled.div`
  display: flex;
  padding: 10px;
  width: 100%;
  overflow: hidden;
`;

export const ThumbnailPreview = styled.div`
  width: 20%;
  position: relative;

  .thumbnail {
    display: block;
    padding-bottom: 85%;
    height: 0;
    margin: 2px;
    background-position: center;
    background-size: cover;
    border-radius: 6px;
  }

  .delete-image {
    position: absolute;
    right: 5px;
    top: 10px;
    font-size: 25px;
    cursor: pointer;
    color: ${colors.strawberry};
    line-height: 0px;
    font-weight: bold;
  }
`;

export const EmojiWrapper = styled.div`
  position: absolute;
  right: 0;
  bottom: 50px;
`;

export const EmojiIconActive = styled(emojiIconActive)`
  position: absolute;
  width: 25px;
  height: 25px;
  right: 10px;
  cursor: pointer;
`;

export const EmojiIconInactive = styled(emojiIconInactive)`
  position: absolute;
  width: 25px;
  height: 25px;
  right: 10px;
  cursor: pointer;
`;

export const Error = styled.span`
  color: ${colors.strawberry};
  font-size: 12px;
`;

const bounceAnimation = keyframes`
  0%, 80%, 100% {
    transform: scale(0); }
  40% {
    transform: scale(1);
  }
`;

export const UploadingImage = styled.div`
  position: absolute;
  top: -2px;
  left: -2px;
  text-align: center;
  padding-top: 35%;
  background-color: rgba(0, 0, 0, 0.8);
  margin: 2px;
  height: 100%;
  width: 100%;
  border-radius: 6px;

  div {
    width: 25px;
    height: 25px;
    background-color: ${colors.steel};
    border-radius: 100%;
    display: inline-block;
    animation: ${bounceAnimation} 1.4s ease-in-out 0s infinite both;
  }

  div:first-child {
    animation-delay: -0.32s;
  }

  div:nth-child(2) {
    animation-delay: -0.16s;
  }
`;
