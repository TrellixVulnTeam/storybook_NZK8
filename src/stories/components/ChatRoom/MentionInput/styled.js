// @flow

import styled, { keyframes } from 'styled-components';
import { colors } from 'utils/StyleGuide';
import { MentionsInput } from 'react-mentions';
import paperclip from './img/paperclip.svg';
import emojiIconActive from './img/emojiIconActive.svg';
import emojiIconInactive from './img/emojiIconInactive.svg';
import everyone from './img/everyone.svg';

export const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  border-top: 1px solid ${colors.paleGrey};
  .custom-mention-input__suggestions {
    width: 100%;
    top: 60px !important;
    left: 0;
    max-height: 400px;
    border-radius: 4px;
    box-shadow: 0 2px 8px 0 rgba(0, 0, 0, 0.1);
    overflow-y: scroll;
  }
`;

export const TextArea = styled(MentionsInput)`
  &.custom-mention-input {
    width: 100%;
    min-height: 80px;
    textarea {
      flex: 1;
      font-size: 14px;
      color: ${colors.charcoalGrey};
      padding: 20px 45px;
      resize: none;
      overflow: hidden;

      ::placeholder {
        color: ${colors.steel};
      }
      &:focus {
        outline: none;
      }
    }
  }
`;

export const everyoneSVG = styled(everyone)`
  width: 40px;
  height: 40px;
`;

export const MentionAreaAvatar = styled.div`
  display: inline-flex;
  ${everyoneSVG} {
    width: 40px;
    height: 40px;
  }
`;

export const MentionAreaAvatarInitials = styled.span`
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: flex;
  background-color: ${colors.shamrockGreen};
  justify-content: center;
  align-items: center;
  color: ${colors.white};
  font-size: 14px;
  font-weight: 700;
`;

export const MentionAreaAvatarImage = styled.span`
  width: 40px;
  height: 40px;
  border-radius: 50%;
  img {
    max-width: 100%;
    object-fit: cover;
    border-radius: 50%;
    width: 40px;
    height: 40px;
  }
`;

export const MentionAreaInfo = styled.div`
  margin-left: 12px;
  flex-flow: column;
  p {
    margin: 0;
    &:first-of-type {
      font-size: 14px;
      font-weight: 600;
    }
    &:nth-of-type(2) {
      font-size: 12px;
    }
    &:nth-of-type(3) {
      font-size: 12px;
      color: ${colors.slateGrey};
    }
  }
`;

export const MentionArea = styled.div`
  display: flex;
  padding: 10px 20px;
  &:hover,
  &:focus {
    background-color: ${colors.brightBlue};
    p {
      color: ${colors.white};
    }
  }
`;

export const FileButton = styled.div`
  position: absolute;
  left: 10px;
  overflow: hidden;
  cursor: pointer;
  top: 20px;
  z-index: 1;

  input[type='file'] {
    position: absolute;
    top: 0;
    right: 0;
    min-width: 100%;
    min-height: 100%;
    font-size: 100px;
    text-align: right;
    filter: alpha(opacity=0);
    opacity: 0;
    outline: none;
    background: white;
    cursor: inherit;
    display: block;
  }
`;

export const PaperclipIcon = styled(paperclip)`
  width: 25px;
  height: 25px;
`;

export const ThumbnailContainer = styled.div`
  display: flex;
  padding: 10px;
  width: 100%;
  overflow: hidden;
`;

export const ThumbnailPreview = styled.div`
  width: 20%;
  position: relative;

  .thumbnail {
    display: block;
    padding-bottom: 85%;
    height: 0;
    margin: 2px;
    background-position: center;
    background-size: cover;
    border-radius: 6px;
  }

  .delete-image {
    position: absolute;
    right: 5px;
    top: 10px;
    font-size: 25px;
    cursor: pointer;
    color: ${colors.strawberry};
    line-height: 0px;
    font-weight: bold;
  }
`;

export const EmojiWrapper = styled.div`
  position: absolute;
  right: 0;
  bottom: 50px;
`;

export const EmojiIconActive = styled(emojiIconActive)`
  position: absolute;
  width: 25px;
  height: 25px;
  right: 10px;
  top: 20px;
  cursor: pointer;
`;

export const EmojiIconInactive = styled(emojiIconInactive)`
  position: absolute;
  width: 25px;
  height: 25px;
  right: 10px;
  top: 20px;
  cursor: pointer;
`;

export const Error = styled.span`
  color: ${colors.strawberry};
  font-size: 12px;
`;

const bounceAnimation = keyframes`
  0%, 80%, 100% {
    transform: scale(0); }
  40% {
    transform: scale(1);
  }
`;

export const UploadingImage = styled.div`
  position: absolute;
  top: -2px;
  left: -2px;
  text-align: center;
  padding-top: 35%;
  background-color: rgba(0, 0, 0, 0.8);
  margin: 2px;
  height: 100%;
  width: 100%;
  border-radius: 6px;

  div {
    width: 25px;
    height: 25px;
    background-color: ${colors.steel};
    border-radius: 100%;
    display: inline-block;
    animation: ${bounceAnimation} 1.4s ease-in-out 0s infinite both;
  }

  div:first-child {
    animation-delay: -0.32s;
  }

  div:nth-child(2) {
    animation-delay: -0.16s;
  }
`;
