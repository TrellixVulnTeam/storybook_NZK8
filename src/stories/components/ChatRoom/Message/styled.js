// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';
import editIcon from 'assets/commonIcons/editIcon.svg';
import trashIcon from 'assets/commonIcons/trashIcon.svg';
import closeIcon from 'assets/commonIcons/closeIcon.svg';
import everyone from './img/everyone.svg';

export const UserTooltip = styled.div`
  display: flex;
  position: absolute;
  height: 70px;
  opacity: 0.85;
  border-radius: 4px;
  box-shadow: 0 2px 20px 0 rgba(0, 0, 0, 0.5);
  background-color: ${colors.darkGrey};
  padding: 10px;
  top: ${p => p.tooltipY}px;
  left: ${p => p.tooltipX}px;
  &:after {
    content: '';
    position: absolute;
    width: 10px;
    height: 10px;
    background: inherit;
    left: 50%;
    bottom: -5px;
    border-radius: 1px;
    transform: rotate(45deg);
  }
`;

export const TooltipAvatarInitials = styled.span`
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: flex;
  background-color: ${colors.shamrockGreen};
  justify-content: center;
  align-items: center;
  color: ${colors.white};
  font-size: 14px;
  font-weight: 700;
`;

export const TooltipAvatar = styled.div`
  display: inline-flex;
`;

export const TooltipInfo = styled.div`
  margin-left: 12px;
  flex-flow: column;

  p {
    font-size: 14px;
    font-weight: 600;
    color: ${colors.white};

    margin: 0;
    &:second-of-type {
      font-size: 13px;
      font-weight: 500;
    }
    &:last-of-type {
      font-size: 12px;
      font-weight: 400;
    }
  }
`;

export const TooltipAvatarImage = styled.div`
  width: 40px;
  height: 40px;
  border-radius: 50%;
  img {
    max-width: 100% !important;
    object-fit: cover;
    border-radius: 50%;
    width: 40px;
    height: 40px;
  }
`;

export const Trash = styled(trashIcon)`
  width: 20px;
  height: 100%;
  margin: 0 auto;
  align-self: center;
`;

export const Edit = styled(editIcon)`
  width: 20px;
  height: 20px;
  margin: 0 auto;
  align-self: center;
  fill: rgb(0, 123, 255);
`;

export const TrashWrapper = styled.div`
  position: absolute;
  display: none;
  overflow: hidden;
  top: 0px;
  right: 0px;
  width: 40px;
  height: 100%;
  background-color: ${colors.babyBlue};
  cursor: pointer;

  div {
    margin: 0 auto;
    align-self: center;
  }

  &:hover {
    background-color: ${colors.dodgerBlue};

    ${Trash} {
      fill: #fff;
    }
  }
`;

export const EditWrapper = styled.div`
  position: absolute;
  display: none;
  overflow: hidden;
  top: 0px;
  right: 40px;
  width: 40px;
  height: 100%;
  background-color: ${colors.babyBlue};
  cursor: pointer;

  div {
    margin: 0 auto;
    align-self: center;
  }

  &:hover {
    background-color: ${colors.dodgerBlue};

    ${Edit} {
      fill: #fff;
    }
  }
`;

export const Close = styled(closeIcon)`
  position: absolute;
  right: 32px;
  width: 24px;
  height: 24px;
  padding: 4px;
  border-radius: 12px;
  background-color: ${colors.alabaster};
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.2);
  fill: ${colors.silver};
  cursor: pointer;

  &:hover {
    fill: #fff;
    background-image: linear-gradient(223deg, ${colors.coral}, ${colors.lipstick});
  }
`;

export const Checkmark = styled.div`
  position: absolute;
  right: 4px;
  width: 24px;
  height: 24px;
  padding: 4px;
  border-radius: 12px;
  background-image: linear-gradient(223deg, #00aeff, #0076ff);
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.2);
  cursor: pointer;

  &:after {
    content: '';
    display: block;
    width: 6px;
    height: 14px;
    border: solid #fff;
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
    margin-left: 4px;
    margin-top: -1px;
  }
`;

export const Wrapper = styled.div`
  position: relative;
  padding: 14px;
  margin: 4px 0;
  background-color: ${p => {
    if (p.editMode) {
      return colors.iceBlue;
    }
    if (p.myMessage && p.type !== null) {
      return colors.lightGrey;
    }
    return 'transparent';
  }};
  border: 1px solid
    ${p => {
      if (p.editMode) {
        return colors.brightBlue;
      }
      if (p.isActivity || (p.type === 'user' && !p.myMessage) || p.type === 'public_share') {
        return colors.paleGrey;
      }
      return 'transparent';
    }};
  border-radius: 6px;
  width: ${p => (p.isActivity ? '100%' : 'auto')};

  &:hover {
    background-color: ${p => (p.myMessage && p.type !== null ? colors.iceBlue : 'transparent')};
    border: 1px solid
      ${p => (p.myMessage && p.type !== null ? colors.brightBlue : colors.lightGrey)};

    ${TrashWrapper} {
      display: flex;
    }

    ${EditWrapper} {
      display: flex;
    }
  }
`;

export const InnerWrapper = styled.div`
  display: flex;
  height: 100%;
  flex-flow: row nowrap;
  justify-content: flex-start;
  align-items: flex-start;
`;

export const Avatar = styled.img`
  display: block;
  width: 40px;
  height: 40px;
  border-radius: 100%;
`;

export const Detail = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  height: 100%;
  padding-left: 18px;
`;

export const Title = styled.div`
  display: flex;
  flex-flow: row nowrap;
  align-items: flex-start;
`;

export const Name = styled.div`
  font-size: 15px;
  font-weight: 600;
  color: ${colors.slateGrey};
`;

export const SubTitle = styled.div`
  padding-left: 20px;
  font-size: 12px;
  color: ${colors.steel};
`;

export const Date = styled.div`
  padding-left: 20px;
  font-size: 12px;
  font-style: italic;
  color: ${colors.steel};
`;

export const Body = styled.div`
  padding-top: 4px;
  color: ${p => (p.isActivity ? colors.slateGrey : colors.black)};
  font-size: ${p => (p.isActivity ? 16 : 14)}px;
  text-align: ${p => (p.isActivity ? 'center' : 'left')};
  white-space: pre-line;
  word-break: break-word;
  position: relative;
  img {
    max-width: 20%;
  }
`;

export const Input = styled.textarea`
  display: block;
  width: 100%;
  height: 100%;
  border-radius: 6px;
  border: 1px solid ${colors.brightBlue};
  background-color: ${colors.lightGrey};

  &:focus {
    box-shadow: none;
    outline: none;
  }
`;

export const everyoneSVG = styled(everyone)`
  width: 40px;
  height: 40px;
  path {
    fill: ${colors.white};
  }
`;
