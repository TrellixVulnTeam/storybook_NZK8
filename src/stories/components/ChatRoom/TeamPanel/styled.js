// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';
import trashIcon from 'assets/commonIcons/trashIcon.svg';
import addIcon from './img/addIcon.svg';

export const Trash = styled(trashIcon)`
  width: 20px;
  margin: 0 auto;
  align-self: center;
`;

export const TrashWrapper = styled.div`
  position: absolute;
  display: none;
  overflow: hidden;
  top: 0px;
  right: 0px;
  width: 40px;
  height: 100%;
  background-color: ${colors.babyBlue};
  cursor: pointer;

  div {
    margin: 0 auto;
    align-self: center;
  }

  :hover {
    background-color: ${colors.dodgerBlue};

    ${Trash} {
      fill: #fff;
    }
  }
`;

export const AddIcon = styled(addIcon)`
  width: 30px;
  height: 30px;
  margin: 8px 20px 8px 9px;
  fill: ${colors.brightBlue};

  &:hover {
    cursor: pointer;
  }
`;

export const Wrapper = styled.li`
  &:hover,
  &:hover label {
    cursor: ${p => (p.disablePointer ? 'auto' : 'pointer')};
  }

  &:hover {
    background-color: ${colors.lightGrey};

    ${TrashWrapper} {
      display: flex;
    }
  }

  margin: 0;
  padding: 18px 20px 18px 18px;
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  align-items: center;
  background-image: ${p => (p.active ? 'linear-gradient(253deg, #4da2ff, #0076ff)' : 'none')};
`;
