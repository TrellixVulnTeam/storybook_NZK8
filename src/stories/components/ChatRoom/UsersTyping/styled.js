// @flow
/* eslint-disable import/prefer-default-export */

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const Wrapper = styled.div`
  padding: 0 20px 10px 20px;
  font-size: 12px;
  color: ${colors.steel};
`;
