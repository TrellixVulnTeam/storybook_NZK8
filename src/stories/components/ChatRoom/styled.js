// @flow

import styled from 'styled-components';
import { lighten } from 'polished';
import { colors } from 'utils/StyleGuide';

export const Wrapper = styled.div`
  position: fixed;
  left: 0;
  top: 0;
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  background-color: ${colors.white};

  @media (min-width: 700px) {
    position: relative;
    width: auto;
    left: auto;
    top: auto;
  }
`;

export const LoaderWrapper = styled.div`
  flex: 1;
  position: absolute;
  width: 100%;
  margin-top: 124px;
`;

export const MessagesListWrapper = styled.div`
  padding: 20px;
  flex: 1;
  overflow-y: auto;

  /* scrollbar */
  ::-webkit-scrollbar {
    width: 7px;
  }
  ::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: #4d5259;
  }
  &:hover {
    ::-webkit-scrollbar-thumb {
      background-color: ${lighten(0.2, '#4d5259')};
    }
  }
`;

export const MessagesList = styled.div`
  display: flex;
  flex-direction: column-reverse;
  align-items: flex-start;
`;

export const Spacer = styled.div`
  height: 20px;
`;
