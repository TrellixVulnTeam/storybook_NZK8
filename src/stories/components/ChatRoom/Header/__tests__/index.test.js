import React from 'react';
import { mount } from 'enzyme';
import ChatRoomHeader from 'components/ChatRoom/Header/index';
import { BackIcon } from 'components/ChatRoom/Header/styled';

describe('<ChatRoomHeader />', () => {
  const historyPushSpy = jest.fn();
  const mandatoryProps = {
    onOpenTeamPanel: jest.fn(),
    session: {
      id: 1,
    },
    history: {
      push: historyPushSpy,
    },
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly with mandatory props', () => {
    const sut = mount(<ChatRoomHeader {...mandatoryProps} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with full props', () => {
    const props = {
      ...mandatoryProps,
      channel: {
        users: [
          {
            id: 0,
            firstName: 'Name',
            lastName: 'Last',
          },
          {
            id: 1,
            firstName: 'Name',
            lastName: 'Last',
          },
          {
            id: 2,
            firstName: 'Name',
            lastName: 'Last',
          },
        ],
        type: 'DIRECT_MESSAGE',
      },
    };
    const sut = mount(<ChatRoomHeader {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with full props and unnamed users', () => {
    const props = {
      ...mandatoryProps,
      channel: {
        users: [
          {
            id: 0,
          },
          {
            id: 1,
          },
          {
            id: 2,
          },
        ],
        type: 'DIRECT_MESSAGE',
      },
    };
    const sut = mount(<ChatRoomHeader {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should redirect after click', () => {
    const sut = mount(<ChatRoomHeader {...mandatoryProps} />);

    expect(historyPushSpy).toHaveBeenCalledTimes(0);

    sut.find(BackIcon).simulate('click');

    expect(historyPushSpy).toHaveBeenCalledTimes(1);
  });
});
