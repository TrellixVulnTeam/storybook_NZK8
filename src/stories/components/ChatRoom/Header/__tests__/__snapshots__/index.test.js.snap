// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatRoomHeader /> should render correctly with full props 1`] = `
<_default
  channel={
    Object {
      "type": "DIRECT_MESSAGE",
      "users": Array [
        Object {
          "firstName": "Name",
          "id": 0,
          "lastName": "Last",
        },
        Object {
          "firstName": "Name",
          "id": 1,
          "lastName": "Last",
        },
        Object {
          "firstName": "Name",
          "id": 2,
          "lastName": "Last",
        },
      ],
    }
  }
  history={
    Object {
      "push": [MockFunction],
    }
  }
  onOpenTeamPanel={[MockFunction]}
  session={
    Object {
      "id": 1,
    }
  }
>
  <styled__Wrapper>
    <div
      className="styled__Wrapper-sc-6yhjoj-0 hvqzLc"
    >
      <styled__BackIcon
        onClick={[Function]}
      >
        <test-file-stub
          className="styled__BackIcon-sc-6yhjoj-3 ftzSuc"
          onClick={[Function]}
        />
      </styled__BackIcon>
      <styled__Avatar
        src="/static/web/assets/defaultAvatar.png"
      >
        <img
          className="styled__Avatar-sc-1wjrk8z-1 bvoqYD"
          src="/static/web/assets/defaultAvatar.png"
        />
      </styled__Avatar>
      <styled__Details>
        <div
          className="styled__Details-sc-6yhjoj-1 emaCwN"
        >
          <label>
            Name Last
          </label>
          <label />
        </div>
      </styled__Details>
      <styled__Options>
        <div
          className="styled__Options-sc-6yhjoj-2 cqAAHz"
        >
          <_default
            value="Team"
          >
            <styled__Wrapper
              data-tip="Team"
              value="Team"
            >
              <div
                className="styled__Wrapper-funpqb-0 cznzQQ"
                data-tip="Team"
                value="Team"
              >
                <span
                  onClick={[MockFunction]}
                >
                  <test-file-stub />
                </span>
                <styled__TooltipWrapper
                  display="true"
                >
                  <div
                    className="styled__TooltipWrapper-funpqb-1 gQtIVd"
                    display="true"
                  >
                    <ReactTooltip
                      clickable={false}
                      delayShow={300}
                      effect="solid"
                      getContent={[Function]}
                      html={true}
                      insecure={true}
                      multiline={true}
                      place="top"
                      resizeHide={true}
                      type="dark"
                      wrapper="div"
                    >
                      <div
                        className="__react_component_tooltip place-top type-dark"
                        data-id="tooltip"
                      >
                        null
                      </div>
                    </ReactTooltip>
                  </div>
                </styled__TooltipWrapper>
              </div>
            </styled__Wrapper>
          </_default>
        </div>
      </styled__Options>
    </div>
  </styled__Wrapper>
</_default>
`;

exports[`<ChatRoomHeader /> should render correctly with full props and unnamed users 1`] = `
<_default
  channel={
    Object {
      "type": "DIRECT_MESSAGE",
      "users": Array [
        Object {
          "id": 0,
        },
        Object {
          "id": 1,
        },
        Object {
          "id": 2,
        },
      ],
    }
  }
  history={
    Object {
      "push": [MockFunction],
    }
  }
  onOpenTeamPanel={[MockFunction]}
  session={
    Object {
      "id": 1,
    }
  }
>
  <styled__Wrapper>
    <div
      className="styled__Wrapper-sc-6yhjoj-0 hvqzLc"
    >
      <styled__BackIcon
        onClick={[Function]}
      >
        <test-file-stub
          className="styled__BackIcon-sc-6yhjoj-3 ftzSuc"
          onClick={[Function]}
        />
      </styled__BackIcon>
      <styled__Avatar
        src="/static/web/assets/defaultAvatar.png"
      >
        <img
          className="styled__Avatar-sc-1wjrk8z-1 bvoqYD"
          src="/static/web/assets/defaultAvatar.png"
        />
      </styled__Avatar>
      <styled__Details>
        <div
          className="styled__Details-sc-6yhjoj-1 emaCwN"
        >
          <label />
          <label />
        </div>
      </styled__Details>
      <styled__Options>
        <div
          className="styled__Options-sc-6yhjoj-2 cqAAHz"
        >
          <_default
            value="Team"
          >
            <styled__Wrapper
              data-tip="Team"
              value="Team"
            >
              <div
                className="styled__Wrapper-funpqb-0 cznzQQ"
                data-tip="Team"
                value="Team"
              >
                <span
                  onClick={[MockFunction]}
                >
                  <test-file-stub />
                </span>
                <styled__TooltipWrapper
                  display="true"
                >
                  <div
                    className="styled__TooltipWrapper-funpqb-1 gQtIVd"
                    display="true"
                  >
                    <ReactTooltip
                      clickable={false}
                      delayShow={300}
                      effect="solid"
                      getContent={[Function]}
                      html={true}
                      insecure={true}
                      multiline={true}
                      place="top"
                      resizeHide={true}
                      type="dark"
                      wrapper="div"
                    >
                      <div
                        className="__react_component_tooltip place-top type-dark"
                        data-id="tooltip"
                      >
                        null
                      </div>
                    </ReactTooltip>
                  </div>
                </styled__TooltipWrapper>
              </div>
            </styled__Wrapper>
          </_default>
        </div>
      </styled__Options>
    </div>
  </styled__Wrapper>
</_default>
`;

exports[`<ChatRoomHeader /> should render correctly with mandatory props 1`] = `
<_default
  history={
    Object {
      "push": [MockFunction],
    }
  }
  onOpenTeamPanel={[MockFunction]}
  session={
    Object {
      "id": 1,
    }
  }
>
  <styled__Wrapper>
    <div
      className="styled__Wrapper-sc-6yhjoj-0 hvqzLc"
    >
      <styled__BackIcon
        onClick={[Function]}
      >
        <test-file-stub
          className="styled__BackIcon-sc-6yhjoj-3 ftzSuc"
          onClick={[Function]}
        />
      </styled__BackIcon>
      <styled__Participants>
        <div
          className="styled__Participants-sc-1wjrk8z-2 koGZMK"
        />
      </styled__Participants>
      <styled__Details>
        <div
          className="styled__Details-sc-6yhjoj-1 emaCwN"
        >
          <label />
          <label />
        </div>
      </styled__Details>
      <styled__Options>
        <div
          className="styled__Options-sc-6yhjoj-2 cqAAHz"
        >
          <_default
            value="Team"
          >
            <styled__Wrapper
              data-tip="Team"
              value="Team"
            >
              <div
                className="styled__Wrapper-funpqb-0 cznzQQ"
                data-tip="Team"
                value="Team"
              >
                <span
                  onClick={[MockFunction]}
                >
                  <test-file-stub />
                </span>
                <styled__TooltipWrapper
                  display="true"
                >
                  <div
                    className="styled__TooltipWrapper-funpqb-1 gQtIVd"
                    display="true"
                  >
                    <ReactTooltip
                      clickable={false}
                      delayShow={300}
                      effect="solid"
                      getContent={[Function]}
                      html={true}
                      insecure={true}
                      multiline={true}
                      place="top"
                      resizeHide={true}
                      type="dark"
                      wrapper="div"
                    >
                      <div
                        className="__react_component_tooltip place-top type-dark"
                        data-id="tooltip"
                      >
                        null
                      </div>
                    </ReactTooltip>
                  </div>
                </styled__TooltipWrapper>
              </div>
            </styled__Wrapper>
          </_default>
        </div>
      </styled__Options>
    </div>
  </styled__Wrapper>
</_default>
`;
